packages/trader/src/Stores/Modules/Trading/Constants/validation-rules.ts

```diff
@@ -6,229 +6,211 @@ import type { TRuleOptions } from 'Utils/Validator/validator';
+                        store.contract_start_type === 'spot' || isHourValid(value ?? ''),
+                    message: localize('Hour must be between 0 and 23.'),
+                },
+            ],
+            [
+                'custom',
+                {
+                    func: (value: TTradeStore['expiry_time'], options: Partial<TRuleOptions>, store: TTradeStore) =>
+                        store.contract_start_type === 'spot' || isMinuteValid(value ?? ''),
+                    message: localize('Minute must be between 0 and 59.'),
+                },
+            ],
+            [
+                'custom',
+                {
+                    func: (value: TTradeStore['expiry_time'], options: Partial<TRuleOptions>, store: TTradeStore) => {
+                        if (store.contract_start_type === 'spot') return true;
+                        if (!isTimeValid(value ?? '')) return false;
+                        const start_moment = toMoment(store.start_date);
+                        const start_moment_clone = start_moment.clone();
+                        const [h, m] = value?.split(':') ?? [];
+                        return isSessionAvailable(store.sessions, start_moment_clone.hour(+h).minute(+m), start_moment);
+                    },
+                    message: localize('Expiry time cannot be in the past.'),
+                },
+            ],
+        ],
+    },
+    ...getMultiplierValidationRules(),
+});
 
-export const getMultiplierValidationRules = () =>
-    ({
-        stop_loss: {
-            rules: [
-                [
-                    'req',
-                    {
-                        condition: (store: TTradeStore) => store.has_stop_loss && !store.stop_loss,
-                        message: localize('Please enter a stop loss amount.'),
-                    },
-                ],
-            ],
-        },
-        take_profit: {
-            rules: [
-                [
-                    'req',
-                    {
-                        condition: (store: TTradeStore) => store.has_take_profit && !store.take_profit,
-                        message: localize('Please enter a take profit amount.'),
-                    },
-                ],
+export const getMultiplierValidationRules = () => ({
+    stop_loss: {
+        rules: [
+            [
+                'req',
+                {
+                    condition: (store: TTradeStore) => store.has_stop_loss && !store.stop_loss,
+                    message: localize('Please enter a stop loss amount.'),
+                },
+            ],
+        ],
+    },
+    take_profit: {
+        rules: [
+            [
+                'req',
+                {
+                    condition: (store: TTradeStore) => store.has_take_profit && !store.take_profit,
+                    message: localize('Please enter a take profit amount.'),
+                },
             ],
-        },
-    } as const);
```

Comment:
removing `as const` that makes the returned object read-only because the type of the returned object is used in trade-store for validation_rules property which should not be read-only, it's mutated in trade-store.